src/backend/services/request_validators.py

```diff
@@ -174,8 +174,22 @@ async def validate_update_agent_request(request: Request):
     Raises:
         HTTPException: If the request does not have the appropriate values in the body
     """
-    body = await request.json()
+    agent_id = request.path_params.get("agent_id")
+    if not agent_id:
+        raise HTTPException(status_code=400, detail="Agent ID is required.")
+
+    agent = agent_crud.get_agent_by_id(session, agent_id)
+    if not agent:
+        raise HTTPException(
+            status_code=400, detail=f"Agent with ID {agent_id} not found."
+        )
 
+    if agent.user_id != request.headers.get("User-Id"):
```

Comment:
This should probably use the `get_header_user_id` method so it works with auth

Reply:
good call out. also changing it for the `validate_user_header` validator as well.

Reply:
let's change it back for `validate_user_header`